### PR TITLE
Make wallet-interface use generic Addresses and Key handling

### PR DESCRIPTION
**Pull request by placer14**
Work in progress tracking #1063 .

### CODE REVIEW

File: core/completion.go [hunk 1]
```diff
@@ -318,13 +311,19 @@ func (n *OpenBazaarNode) ReleaseFundsAfterTimeout(contract *pb.RicardianContract
...
 			}
+
 			outpoint := wire.NewOutPoint(hash, r.Index)
-			utxo.Op = *outpoint
-			utxos = append(utxos, utxo)
+			var txInput = wallet.TransactionInput{
+				Value:         r.Value,
+				OutpointIndex: outpoint.Index,
+				OutpointHash:  outpoint.Hash.CloneBytes(),
+			}
```
Comment (placer14) [inline core/completion.go]: @cpacia Do we need to specify the `LinkedAddress` here? Whose?

Comment (cpacia) [reply core/completion.go]: Yes I believe so. Should be in the TransactionRecord obj after we migrate it.

Comment (placer14) [reply core/completion.go]: Fixed.

File: core/disputes.go [hunk 2]
```diff
@@ -982,25 +967,23 @@ func (n *OpenBazaarNode) ValidateDisputeResolution(contract *pb.RicardianContrac
...
+}
+
+func (n *OpenBazaarNode) verifyPaymentDestinationIsInWallet(output *pb.DisputeResolution_Payout_Output) error {
+	addr, err := pb.DisputeResolutionPayoutOutputToAddress(n.Wallet, output)
+	if err != nil {
+		return err
+	}
+
+	if !n.Wallet.HasKey(addr) {
+		return errors.New("Moderator dispute resolution payout address is not defined in your wallet to recieve funds")
```
[the remark this answers is not in the document]
Comment (placer14) [reply core/disputes.go]: https://github.com/OpenBazaar/openbazaar-go/pull/1086/files#diff-dd186645aefa3a772c7545050c2cae54R463

[the remark this answers is not in the document]
Comment (placer14) [reply core/disputes.go]: Literally every error in this file is a sentence that's capitalized. Can we return to this later when/if we redo our error handling?

File: core/completion.go [hunk 3]
```diff
@@ -163,12 +162,17 @@ func (n *OpenBazaarNode) CompleteOrder(orderRatings *OrderRatings, contract *pb.
...
 			if !r.Spent && r.Value > 0 {
-				outpointHash, err := hex.DecodeString(r.Txid)
+				addr, err := n.Wallet.DecodeAddress(r.Address)
 				if err != nil {
 					return err
 				}
 				outValue += r.Value
-				in := wallet.TransactionInput{OutpointIndex: r.Index, OutpointHash: outpointHash, Value: r.Value}
+				in := wallet.TransactionInput{
+					LinkedAddress: addr,
```
Comment (placer14) [inline core/completion.go]: Note this `LinkedAddress` was missing in the original struct.

**Pull request merged**